packages/react/src/Form/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { ComponentProps, ReactNode } from "react";
+import { Option } from "../core";
+import { CheckboxItem } from "./CheckboxItem";
+import { CheckboxModes } from "./CheckboxModes";
+
+type OnChange = {
+  onChange: (data: {
+    values: string[];
+    target: { value: string; checked: boolean };
+    name?: string;
+    id?: string;
+  }) => void;
+};
+
+type Props = Omit<
+  ComponentProps<typeof CheckboxItem>,
+  "value" | "label" | "checked" | "onChange"
+> & {
+  options: Option[];
+  values?: string[];
+  children?: ReactNode;
+  mode?: keyof typeof CheckboxModes;
+} & OnChange;
+
+const defaultClassName = (mode: string) => {
+  switch (mode) {
+    case CheckboxModes.default:
+      return "af-form__checkbox-custom";
+    case CheckboxModes.classic:
+      return "af-form__checkbox";
+    default:
+      return `af-form__checkbox-${mode}`;
+  }
+};
+
+const CheckboxInner = ({
+  id,
+  name,
+  options,
+  disabled,
+  children,
+  values = [],
+  mode = CheckboxModes.default,
+  onChange,
+  ...otherProps
+}: Props) => {
+  const className = defaultClassName(mode);
+
+  const handleOnChange: React.ChangeEventHandler<HTMLInputElement> = ({
+    target: { value, checked },
+  }) => {
+    const newValues = checked
+      ? [...values, value]
+      : values.filter((v) => v !== value);
+    onChange({ values: newValues, target: { value, checked }, id, name });
+  };
+  return (
+    <>
+      {options.map((option) => {
+        const isChecked = values ? values.indexOf(option.value) >= 0 : false;
+        return (
+          <CheckboxItem
+            {...otherProps}
+            onChange={handleOnChange}
+            key={option.value}
+            className={className}
+            id={option.id}
+            value={option.value}
+            label={option.label}
+            isChecked={isChecked}
+            name={name}
+            disabled={option.disabled || disabled}
+          >
```

Comment:
```suggestion
          <CheckboxItem
            {...otherProps}
            {...option}
            onChange={handleOnChange}
            key={option.value}
            className={className}
            isChecked={isChecked}
            name={name}
            disabled={option.disabled || disabled}
          >
```
Possible de déstructurer `option` dans `CheckboxItem` ?

Reply:
perso, pour 3 champs je suis pas fan. Ca masque un peu ce qui est fait. Mais si vous êtes sûrs de votre suggestion je changerai

Reply:
ben option d'après le typage n'a que 3 clé id, label, value et disabled
https://github.com/AxaFrance/design-system/blob/main/packages/react/src/Form/core/index.ts#L18

donc pour moi la proposition est valide

Reply:
@pplancq j'ai pas compris ton message.
Oui il y a 3 champs, c'est ce que j'ai dit dans mon commentaire. Et j'ai plutot tendence à ne pas utiliser le spread dans un composant, car dans un environnement sans typage (comme github), on ne sait pas ce qui est spread. Donc comme 3 champs c'est pas beaucoup, je préfère ne pas spread, comme ça on voit bien ce qu'on passe au composant.
C'est plus clair pour toi ?

Reply:
j'ai changé pour le spread, je clos.

Reply:
Ben moi l'un ou l'autre me va, après c'est vrai que coté lisibilité c'est pas ouf.
Après comme on est en typescript je fais confiance a son analyse, s'il dit qu'il n'y a pas d'erreur pour moi c'est que c'est bon et qu'il y a bien des props valide e que donc le spread marche.